src/snippets/codemirror/snippet_change_spec.ts

```diff
@@ -25,35 +25,48 @@ export class SnippetChangeSpec {
             if (!(text.charAt(i) === "$")) {
                 continue;
             }
-
+    
             let number:number = parseInt(text.charAt(i + 1));
-
+    
             const tabstopStart = i;
             let tabstopEnd = tabstopStart + 2;
             let tabstopReplacement = "";
-
-
+    
+    
             if (isNaN(number)) {
-                // Check for selection tabstops of the form ${0:XXX}
-                if (!(text.charAt(i+1) === "{" && text.charAt(i+3) === ":")) continue;
-
-                number = parseInt(text.charAt(i + 2));
+                // Check for selection tabstops of the form ${\d+:XXX} where \d+ is some number of
+                // digits and XXX is the replacement string, separated by a colon
+                if (!(text.charAt(i+1) === "{")) continue;
+    
+                // Find the index of the matching closing bracket
+                const closingIndex = text.indexOf("}", i)
```

Comment:
You should use the `findMatchingBracket` utility function here because placeholder replacement strings can contain `{}`.